app/Services/FetchRepositoriesService.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use App\DTO\OrganizationDTO;
+use App\Exceptions\UnableToMakeDtoFromResponseException;
+use App\Http\Integrations\GithubConnector;
+use App\Http\Integrations\Requests\GetRepositoriesRequest;
+use App\Models\Organization;
+use App\Models\Repository;
+use App\Models\User;
+use Exception;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\UnauthorizedException;
+
+class FetchRepositoriesService
+{
+    public function __construct(
+        protected GithubConnector $githubConnector,
+    ) {}
+
+    public function fetchRepositories(OrganizationDTO $organizationDto): void
+    {
+        $organization = Organization::query()->where("github_id", $organizationDto->githubId)->firstOrFail();
+        $user = User::query()->where("id", Auth::user()->id)->firstOrFail();
+
+        if ($user->organizations()
+            ->where("organization_id", $organization->id)
+            ->where("is_admin", true)
+            ->exists()) {
+            $request = new GetRepositoriesRequest($organizationDto);
+
+            $response = $this->githubConnector->send($request);
+
+            try {
+                $this->storeRepositories($response->dto());
```

Comment:
```suggestion
            try {
                $request = new GetRepositoriesRequest($organizationDto);

                $response = $this->githubConnector->send($request);

                $this->storeRepositories($response->dto());
```
For handling github connection errors

Reply:
Github connector has `AlwaysThrowOnErrors` trait and throws appropriate exceptions by itself, is there a need to include it in try catch?

Reply:
included in try catch and throwing custom exception